src/frontend/src/binder/expr/mod.rs

```diff
@@ -378,6 +378,14 @@ impl Binder {
     }
 
     fn bind_parameter(&mut self, index: u64) -> Result<ExprImpl> {
+        // Special check for sql udf
+        // Note: This is specific to anonymous sql udf, since the
+        // parameters will be parsed and treated as `Parameter`.
+        // For detailed explanation, consider checking `bind_column`.
+        if let Some(expr) = self.udf_context.get(&format!("${index}")) {
```

Comment:
What if we use sql-udf in extended mode with parameter? 😕

Reply:
This PR acts as the backbone for subsequent sql udf implementations, and specifically supports anonymous sql udf, naming related logic will be resolved in later PRs.
Though not implementing yet, the logic would be pretty much the same, we will try grab the identifier key of parameter in the context, to see if this is a parameter of sql udf instead of a normal column.

Reply:
I'm just wondering if it's possible to make it a graceful "not implemented" error when combining the usage of extended mode and SQL UDF, **right after** this PR. Binding to incorrect results will be confusing.

Reply:
>  if it's possible to make it a graceful "not implemented" error when combining the usage of extended mode and SQL UDF

Yep this is possible, I'll come up with the PR that will gracefully tell the users that the use like,
```sql
create function with_param(a INT, b INT) returns int language sql as 'select a + b';
```
is not implemented yet, right after the merge of this one.

Reply:
Oops. I guess I didn't express it clearly. I mean the parameter in prepared statements.

https://arc.net/l/quote/lgdgssbp